File: echo.rb
# frozen_string_literal: true

require 'sinatra'

class Echo < Sinatra::Base
  get '/' do
    redirect 'https://docs.cloudwalk.io'
  end

  post '/' do
    sleep cooldown

    if params['anotherVariable']
      "CLOUDWALK #{params['buf']} #{params['anotherVariable']}"
    else
      "CLOUDWALK #{params['buf']}"
    end
  end

  helpers do
    def cooldown
      s = params['sleep'].to_f
      s > 2.0 ? 2.0 : s
    end
  end
end

File: echo.go
package main

import (
	"fmt"
	"net/http"

	"google.golang.org/appengine"
)

func handleEcho(w http.ResponseWriter, r *http.Request) {
	if r.Method == "POST" {
		if r.FormValue("anotherVariable") != "" {
			fmt.Fprintf(w, fmt.Sprintf("CLOUDWALK %s %s", r.FormValue("buf"), r.FormValue("anotherVariable")))
		} else {
			fmt.Fprintf(w, fmt.Sprintf("CLOUDWALK %s", r.FormValue("buf")))
		}
		return
	}
	http.Redirect(w, r, "https://docs.cloudwalk.io", http.StatusSeeOther)
}

func init() {
	http.HandleFunc("/", handleEcho)
}

func main() {
	appengine.Main()
}
